packages/yew/src/virtual_dom/vnode.rs

```diff
@@ -45,8 +46,23 @@ impl VNode {
         }
     }
 
+    /// Returns the first DOM node if available
+    pub(crate) fn first_node(&self) -> Option<Node> {
+        match self {
+            VNode::VTag(vtag) => vtag.reference().cloned().map(JsCast::unchecked_into),
+            VNode::VText(vtext) => vtext
+                .reference
+                .as_ref()
+                .cloned()
+                .map(JsCast::unchecked_into),
+            VNode::VComp(vcomp) => vcomp.node_ref.get(),
+            VNode::VList(vlist) => vlist.get(0).and_then(VNode::first_node),
+            VNode::VRef(node) => Some(node.clone()),
```

Comment:
`VNode::VList(vlist) => vlist.get(0).and_then(VNode::first_node)`

would vlist be empty and cause out of index range error?

Reply:
vlist.get() is the same as Vec::get, so it returns `Option::None` when the index is out of bounds.